pkg/block/fetcher.go

```diff
@@ -566,4 +573,51 @@ func (f *ConsistencyDelayMetaFilter) Filter(metas map[ulid.ULID]*metadata.Meta,
 			delete(metas, id)
 		}
 	}
+
+	return nil
+}
+
+// IgnoreDeletionMarkFilter is a MetaFetcher that filters blocks that are marked for deletion; delay duration before current time.
+// The delay duration is to make sure that the replacement block can be fetched before we filter out the old block.
```

Comment:
```suggestion
// The delay duration is to make sure that the replacement block can be fetched before we filter out the old block. Delay is not considered when computing DeletionMarkBlocks map.
//
// Not go-routine safe. 
```
Can we add `Not go routine safe` everywhere else in this file (just structs)

Reply:
Everything runs in a goroutine, so "Not go routine safe" doesn't make sense. Not safe for concurrent usage from multiple goroutines?

Reply:
Indeed it is called "thread safe" probably in most places e.g. https://www.gnu.org/software/libc/manual/html_node/POSIX-Safety-Concepts.html in POSIX. So, `goroutine safe` is a Go version of the same phrase :p

Reply:
I have added `Not go-routine safe.` above all structs excluding `syncMetrics` struct.